pkg/codegen/pcl/binder_resource.go

```diff
@@ -330,6 +330,8 @@ func (b *binder) bindResourceBody(node *Resource) hcl.Diagnostics {
 				case "ignoreChanges":
 					t = model.NewListType(ResourcePropertyType)
 					resourceOptions.IgnoreChanges = item.Value
+				case "version":
+					continue
```

Comment:
Shouldn't we assert that this is a string.
We should probably do the same for `pluginDownloadURL` (which always appears in the same context as `version`, since (name, version, pluginDownloadURL) is the triple used to uniquely define a provider.)

Reply:
This PR is mostly intended to unblock[ PR 291](https://github.com/pulumi/pulumi-yaml/pull/291), as we'll implement the logic for the binder to handle versions in a separate PR. Right now, it is still being ignored, so we could do the assertion in the separate PR?